enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+   
+   b. The installer then sets the ingress cluster object with the `eipAllocations`.
+   
+   c. The cluster ingress operator then creates a default ingress controller by:
+
+     i.   creating an default ingress controller CR
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eipAllocations` from the
+          ingress cluster object.
+     
+     iii. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+          which uses the value from the field `eipAllocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CR:
+   
+   The admin sets the `eipAllocations` in the Ingress Controller CR. Then same process
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
+   
+   When the admin changes the eipAllocations field in the Ingress Controller CR, the ingress operator sets the Load Balancer
+   Progressing status as per [section](#Cluster administrator updates eipAllocations field of the existing ingress controller)
+   Then step 1.c.iii is followed by the ingress operator.
+
+   If the admin wants to detect the change in the eipAllocations and update the LB service automatically.
+   rather than deleting the Service explicitly, it is possible to annotate the IngressController with
+   the newly defined ingress.operator.openshift.io/auto-delete-load-balancer annotation. 
+   If the operator observes that this annotation is set and that its 
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations` field has been changed, 
+   the operator automatically deletes the Service if needed to complete an eip change operation. 
+   This purpose of this annotation is to simplify automation: A tool can set the annotation when it 
+   creates the IngressController or when it updates `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations` 
+   to instruct the operator to perform the operation automatically. This annotation is not intended for
+   end-users to use directly.
+   
+### Risks and Mitigations
+
+- Providing a right number of EIPs same as the number of public subnets as per AWS guidelines is an admin
+responsibility.
+- Also providing existing available EIPs which are not assigned to any instance is important.
+- If the above two points are not followed, the kubernetes LoadBalancer service for the Ingress Controller
+won't get assigned EIPs which will cause the LoadBalancer service to be in pending state by the 
+(CCM)Cloud Controller Manager. The reason for the error could be seen in the Status of the Ingress Controller.
+As of now we are not able to identify any more risks which can affect other components of the EP.
+
+### Drawbacks
+
+Currently, there are not any drawbacks as such for this EP.
```

Comment:
I think we could add some draw backs:

 ```suggestion
Requiring the cluster admin to delete the LoadBalancer-type
service leads to several minutes of ingress traffic disruption.

This enhancement brings additional engineering complexity for upgrade
scenarios because cluster admins have previously been allowed to directly
add this annotation on a service. 

Debugging invalid EIP Allocations will be confusing for cluster admins and may
lead to extra support cases or bugs.

This design requires a cluster admin to check the IngressController's status after
updating `eipAllocations` for instructions on how to proceed.
```

Reply:
ok